airbyte-integrations/connectors/source-google-ads/source_google_ads/source.py

```diff
@@ -39,13 +39,52 @@
     AdGroupAds,
     AdGroups,
     Campaigns,
+    CustomQuery,
+    CustomQueryFullRefresh,
+    CustomQueryIncremental,
     DisplayKeywordPerformanceReport,
     DisplayTopicsPerformanceReport,
     ShoppingPerformanceReport,
 )
 
 
 class SourceGoogleAds(AbstractSource):
+    def get_local_json_schema(self, config) -> MutableMapping[str, Any]:
+        """
+        As agreed, now it returns the default schema (since read -> schema_generator.py may take hours for the end user).
+        If we want to redesign json schema from raw query, this method need to be modified.
+        """
+        local_json_schema = {"$schema": "http://json-schema.org/draft-07/schema#", "type": "object", "additionalProperties": True}
+        return local_json_schema
+
+    def discover(self, logger: AirbyteLogger, config: Mapping[str, Any]) -> AirbyteCatalog:
+        # streams = [stream.as_airbyte_stream() for stream in self.streams(config=config)]
+        streams = []
+        for stream in self.streams(config=config):
+            if not isinstance(stream, (CustomQueryFullRefresh, CustomQueryIncremental)):
```

Comment:
checking for CustomQuery is enough,
but in general the whole check is not needed, see my other comments